include/vk/vkwrap/surface.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "common/glfw_include.h"
+#include "common/vulkan_include.h"
+
+namespace vkwrap
+{
+
+class Surface : private vk::UniqueSurfaceKHR
+{
+
+  private:
+    using BaseType = vk::UniqueSurfaceKHR;
+
+  private:
+    // Call permissions: any thread.
+    static vk::SurfaceKHR createSurface( vk::Instance instance, GLFWwindow* window );
```

Comment:
Surface should not depend on a window interface. By using GLFWwindow in constructor you made this code non-portable between glfw, sdl e.t.c. As I see it, surface does not need to know how to create itself. Rather, it is the window system responsibility to create a Surface.

Reply:
Yep. Only window system knows, how to communicate with its objects, so it should provide api.